com.liferay.blade.cli/src/com/liferay/blade/cli/DeployCommand.java

```diff
@@ -169,80 +170,91 @@ private void installOrUpdate(File outputFile) throws Exception {
 			isFragment = fragmentHost != null;
 
 			bsn = bundle.getBsn();
+
+			if(isFragment) {
+				hostBSN =
+						new Parameters(fragmentHost).keySet().iterator().next();
```

Comment:
There may be multiple parameters in the hostBSN so we can't asssume the "host" bundle will be the first parameter.